test/core/end2end/tests/retry_streaming.cc

```diff
@@ -59,345 +45,129 @@ static std::unique_ptr<CoreTestFixture> begin_test(
-                                grpc_core::CqVerifier::tag(104), nullptr);
-  GPR_ASSERT(GRPC_CALL_OK == error);
-  cqv.Expect(grpc_core::CqVerifier::tag(104), true);
-  cqv.Verify();
-
+  CoreEnd2endTest::IncomingCloseOnServer client_close;
+  s.NewBatch(104)
+      .RecvCloseOnServer(client_close)
+      .SendInitialMetadata({})
+      .SendStatusFromServer(GRPC_STATUS_ABORTED, "xyz", {});
+  Expect(104, true);
+  Step();
   // Clean up from first attempt.
-  grpc_call_unref(s);
-  grpc_metadata_array_destroy(&request_metadata_recv);
-  grpc_metadata_array_init(&request_metadata_recv);
-  grpc_call_details_destroy(&call_details);
-  grpc_call_details_init(&call_details);
-  GPR_ASSERT(byte_buffer_eq_slice(request_payload_recv, request_payload_slice));
-  grpc_byte_buffer_destroy(request_payload_recv);
-  request_payload_recv = nullptr;
-  GPR_ASSERT(
-      byte_buffer_eq_slice(request2_payload_recv, request2_payload_slice));
-  grpc_byte_buffer_destroy(request2_payload_recv);
-  request2_payload_recv = nullptr;
-
+  EXPECT_EQ(client_message.payload(), "foo");
+  EXPECT_EQ(client_message2.payload(), "bar");
   // Server gets a second call (the retry).
-  error = grpc_server_request_call(f->server(), &s, &call_details,
-                                   &request_metadata_recv, f->cq(), f->cq(),
-                                   grpc_core::CqVerifier::tag(201));
-  GPR_ASSERT(GRPC_CALL_OK == error);
-  cqv.Expect(grpc_core::CqVerifier::tag(201), true);
-  cqv.Verify();
-
-  peer = grpc_call_get_peer(s);
-  GPR_ASSERT(peer != nullptr);
-  gpr_log(GPR_DEBUG, "server_peer=%s", peer);
-  gpr_free(peer);
-  peer = grpc_call_get_peer(c);
-  GPR_ASSERT(peer != nullptr);
-  gpr_log(GPR_DEBUG, "client_peer=%s", peer);
-  gpr_free(peer);
-
+  auto s2 = RequestCall(201);
+  Expect(201, true);
+  Step();
+  EXPECT_NE(s.GetPeer(), absl::nullopt);
+  EXPECT_NE(c.GetPeer(), absl::nullopt);
   // Server receives a message.
-  memset(ops, 0, sizeof(ops));
-  op = ops;
-  op->op = GRPC_OP_RECV_MESSAGE;
-  op->data.recv_message.recv_message = &request_payload_recv;
-  op++;
-  error = grpc_call_start_batch(s, ops, static_cast<size_t>(op - ops),
-                                grpc_core::CqVerifier::tag(202), nullptr);
-  GPR_ASSERT(GRPC_CALL_OK == error);
-  cqv.Expect(grpc_core::CqVerifier::tag(202), true);
-  cqv.Verify();
-
-  // Server receives a second message.
-  memset(ops, 0, sizeof(ops));
-  op = ops;
-  op->op = GRPC_OP_RECV_MESSAGE;
-  op->data.recv_message.recv_message = &request2_payload_recv;
-  op++;
-  error = grpc_call_start_batch(s, ops, static_cast<size_t>(op - ops),
-                                grpc_core::CqVerifier::tag(203), nullptr);
-  GPR_ASSERT(GRPC_CALL_OK == error);
-  cqv.Expect(grpc_core::CqVerifier::tag(203), true);
-  cqv.Verify();
-
+  CoreEnd2endTest::IncomingMessage client_message3;
+  s2.NewBatch(202).RecvMessage(client_message3);
+  Expect(202, true);
+  Step();
+  CoreEnd2endTest::IncomingMessage client_message4;
```

Comment:
Please re-add the "Server receives a second message" comment here.

Reply:
done